### PR TITLE
improve `isBooleanNode()` to fix false negative error in `prefer-array-some` rule

### PR DESCRIPTION
**Pull request by B2o5T**
fix https://github.com/sindresorhus/eslint-plugin-unicorn/issues/996

### CODE REVIEW

File: test/prefer-array-some.mjs [hunk 1]
```diff
@@ -72,20 +77,14 @@ test({
...
-					message: 'Prefer `.some(…)` over `.find(…)`.',
-					suggestions: [
-						{
-							desc: 'Replace `.find(…)` with `.some(…)`.',
-							output: 'if (foo.some(fn)) {}',
-						},
-					],
-				},
-			],
-		},
```
Comment (fisker) [inline test/prefer-array-some.mjs:88]: This test mean to test the actual message, please revert.

File: rules/utils/boolean.js [hunk 2]
```diff
@@ -71,6 +73,26 @@ function isBooleanNode(node) {
 		return isBooleanNode(parent);
 	}
 
+	if (scope && parent.type === 'VariableDeclarator') {
+		const reference = findReference(scope, parent.id);
```
Comment (fisker) [inline rules/utils/boolean.js:77]: The `parent.id` can be other things, not only `Identifier`, we need tests.

File: test/prefer-array-some.mjs (hunk 1 above)
Comment (B2o5T) [reply test/prefer-array-some.mjs:88]: I know, I made a refactoring to use `invalidCase` function instead

Comment (fisker) [reply test/prefer-array-some.mjs:88]: Message not asserted in that function, only messageId.

Comment (B2o5T) [reply test/prefer-array-some.mjs:88]: asserted on line 82

Comment (fisker) [reply test/prefer-array-some.mjs:88]: https://github.com/sindresorhus/eslint-plugin-unicorn/pull/635 This is why we need those tests.

**Review by m0dul3 — APPROVED** [review]

**Review by m0dul3 — APPROVED** [review]

**Review by m0dul3 — APPROVED** [review]

**Review by m0dul3 — APPROVED** [review]

**Comment by sindresorhus** [comment]
@B2o5T Friendly bump

File: rules/utils/boolean.js (hunk 2 above)
Comment (B2o5T) [reply rules/utils/boolean.js:77]: Could you suggest me some tests that I need to add?

Comment (fisker) [reply rules/utils/boolean.js:77]:
```js
const {} = foo;
const [] = foo;
```

File: rules/utils/find-reference.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+'use strict';
+
+/**
+ * Finds the eslint-scope reference in the given scope.
+ * @param {Object} scope The scope to search.
+ * @param {ASTNode} node The identifier node.
+ * @returns {Reference|undefined} Returns the found reference or null if none were found.
+ */
+function findReference(scope, node) {
```
Comment (fisker) [inline rules/utils/find-reference.js:9]: Better to use `getVariable` from `eslint-utils`.

**Comment by sindresorhus** [comment]
@B2o5T Bump

**Comment by B2o5T** [comment]
@sindresorhus @fisker sorry, don’t have a time to finish it, if someone want give a try, feel free

**Pull request closed without merging**